skyportal/handlers/api/news_feed.py

```diff
@@ -44,13 +55,29 @@ def get(self):
               application/json:
                 schema: Error
         """
-        preferences = (
-            self.current_user.preferences if self.current_user.preferences else {}
-        )
-        if 'newsFeed' in preferences and 'numItems' in preferences['newsFeed']:
-            n_items = min(int(preferences['newsFeed']['numItems']), 50)
+
+        if hasattr(self.current_user, 'preferences') and isinstance(
+            self.current_user.preferences, dict
+        ):
+            preferences = self.current_user.preferences
         else:
-            n_items = 10
+            preferences = {}
+
+        n_items = self.get_query_argument("numItems", None)
```

Comment:
This feels too complicated.
How about getting:
1. arg n_items
2. preference n_items
3. choose and compare